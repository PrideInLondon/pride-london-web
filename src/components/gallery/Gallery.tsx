import React, { useState, useEffect } from 'react'
import { GalleryProps } from './Gallery.types'

export const Gallery = <T,>({
  entries,
  render,
  pageSize = entries.length,
}: GalleryProps & {
  entries: T[]
  render: (renderProps: {
    entries: T[]
    moreEntriesToShow: boolean
    showNextPage: () => void
  }) => React.ReactNode
}) => {
  const [numberOfEntriesToShow, setNumberOfEntriesToShow] = useState(pageSize)

  useEffect(() => {
    setNumberOfEntriesToShow(pageSize)
  }, [pageSize])

  return (
    <>
      {render({
        entries: entries.slice(0, numberOfEntriesToShow),
        moreEntriesToShow: numberOfEntriesToShow < entries.length,
        showNextPage: () =>
          setNumberOfEntriesToShow(numberOfEntriesToShow + pageSize),
      })}
    </>
  )
}
