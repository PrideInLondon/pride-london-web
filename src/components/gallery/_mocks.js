export const mockData = [
  {
    name: 'Test image 1',
    originalAlt: 'foo',
    original: 'foo',
    description: 'Test decription',
    thumbnailAlt: 'foo',
    thumbnail: 'foo',
    photographer: {
      name: 'Test photographer 1',
      url: 'example.com',
    },
  },
]
