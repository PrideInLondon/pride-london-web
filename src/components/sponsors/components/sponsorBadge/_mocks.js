export const mockSponsor = {
  level: 'Headline',
  logo: '/logo1',
  name: 'Sponsor One',
  url: 'https://sponsor1.com',
}
