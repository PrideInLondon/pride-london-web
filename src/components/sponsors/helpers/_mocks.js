export const mockSponsors = [
  {
    level: 'Headline',
    logo: '/logo1',
    name: 'Sponsor One',
    url: 'https://sponsor1.com',
  },
  {
    level: 'Silver',
    logo: '/logo2',
    name: 'Sponsor Two',
    url: 'https://sponsor2.com',
  },
  {
    level: 'Silver',
    logo: '',
    name: 'Sponsor Three',
    url: 'https://sponsor3.com',
  },
]
