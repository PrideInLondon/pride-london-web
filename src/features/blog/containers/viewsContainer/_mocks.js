export const views = [
  {
    author: {
      display_name: {
        display_name: 'Test',
      },
    },
    category: 'Views',
    datePublished: '2019-05-17',
    headerImage: {
      file: {
        url: 'src',
      },
    },
    id: '1',
    title: 'test',
  },
  {
    author: {
      display_name: {
        display_name: 'Test',
      },
    },
    category: 'Views',
    datePublished: '2019-05-17',
    headerImage: {
      file: {
        url: 'src',
      },
    },
    id: '2',
    title: 'test',
  },
]
