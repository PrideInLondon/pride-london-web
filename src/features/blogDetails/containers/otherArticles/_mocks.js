export default {
  allContentfulArticle: {
    edges: [
      {
        node: {
          id: '73369e14-9fc7-5bec-a2ed-e3630d062d6b',
          title: 'A Featured Article',
          datePublished: '2019-08-14',
          category: 'Partners',
        },
      },
      {
        node: {
          id: '1a0304c1-9e37-5adc-9e7c-7a67adc8aeba',
          title: 'Hey hooo hola',
          datePublished: '2019-06-11',
          category: 'Partners',
        },
      },
      {
        node: {
          id: 'bf65a5ff-8b99-5d86-b0eb-1dabb58e0208',
          title: 'Yada yada yada',
          datePublished: '2019-05-17',
          category: 'Partners',
        },
      },
    ],
  },
}
