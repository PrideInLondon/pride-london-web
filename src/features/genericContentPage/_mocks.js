export const buttonNode = {
  data: {
    target: {
      fields: {
        buttonContent: { 'en-GB': 'Button primary type' },
        buttonHref: { 'en-GB': 'https://prideinlondon.org/who-we-are' },
        buttonPrimary: { 'en-GB': true },
      },
      sys: {
        contentType: {
          sys: {
            id: 'button',
          },
        },
      },
    },
  },
}

export const videoNode = {
  data: {
    target: {
      fields: {
        title: { 'en-GB': 'Video for example' },
        videoId: { 'en-GB': '7ge36or-nz4' },
      },
      sys: {
        contentType: {
          sys: {
            id: 'video',
          },
        },
      },
    },
  },
}

export const imageNode = {
  data: {
    target: {
      fields: {
        description: {
          'en-GB': 'Image description',
        },
        file: {
          'en-GB': {
            contentType: 'image/jpeg',
            details: {
              image: { width: 5152, height: 3864 },
              size: 8068085,
            },
            fileName: 'image.jpg',
            url: 'src',
          },
        },
      },
    },
  },
}
