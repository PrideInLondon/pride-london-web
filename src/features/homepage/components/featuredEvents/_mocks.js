export const mockEvents = [
  {
    node: {
      id: '123abc',
      name: 'Pride Quiz',
      startTime: '2019-06-05T20:00+01:00',
      endTime: '2019-06-05T23:00+01:00',
      eventPriceLow: 3,
      eventsListPicture: {
        file: {
          url: 'test/image.jpg',
          title: 'image alt',
        },
      },
    },
  },
  {
    node: {
      id: '123abcd',
      name: 'Pride Quiz 2',
      startTime: '2019-06-05T20:00+01:00',
      endTime: '2019-06-05T23:00+01:00',
      eventPriceLow: 3,
      eventsListPicture: {
        file: {
          url: 'test/image.jpg',
          title: 'image alt',
        },
      },
    },
  },
]
