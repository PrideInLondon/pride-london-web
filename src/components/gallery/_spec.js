import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import 'jest-styled-components'
import { mockData } from './_mocks'
import Gallery from './index'

describe('<Gallery/>', () => {
  it('should render and match snapshot', () => {
    const wrapper = shallow(<Gallery images={mockData} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
