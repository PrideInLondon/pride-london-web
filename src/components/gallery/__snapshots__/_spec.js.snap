// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery/> should render and match snapshot 1`] = `
<styles__GalleryContainer>
  <Slider
    arrows={true}
    beforeChange={[Function]}
    centerMode={true}
    centerPadding="15%"
    className="gallery-slick"
    infinite={true}
    responsive={
      Array [
        Object {
          "breakpoint": 767,
          "settings": Object {
            "arrows": false,
            "centerMode": true,
            "centerPadding": "100px",
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
        Object {
          "breakpoint": 550,
          "settings": Object {
            "arrows": false,
            "centerMode": true,
            "centerPadding": "40px",
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={1}
    speed={450}
  >
    <GalleryImage
      description="Test decription"
      key="Test image 1-photo"
      name="Test image 1"
      original="foo"
      originalAlt="foo"
      photoIndex={1}
      photographer={
        Object {
          "name": "Test photographer 1",
          "url": "example.com",
        }
      }
      thumbnail="foo"
      thumbnailAlt="foo"
      totalPhotos={1}
    />
  </Slider>
  <Slider
    arrows={false}
    centerMode={true}
    centerPadding="15%"
    className="gallery-slick"
    draggable={false}
    infinite={true}
    responsive={
      Array [
        Object {
          "breakpoint": 767,
          "settings": Object {
            "arrows": false,
            "centerMode": true,
            "centerPadding": "100px",
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
        Object {
          "breakpoint": 550,
          "settings": Object {
            "arrows": false,
            "centerMode": true,
            "centerPadding": "40px",
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={1}
    speed={450}
  >
    <GalleryImageDetails
      description="Test decription"
      key="Test image 1-description"
      name="Test image 1"
      original="foo"
      originalAlt="foo"
      photographer={
        Object {
          "name": "Test photographer 1",
          "url": "example.com",
        }
      }
      thumbnail="foo"
      thumbnailAlt="foo"
    />
  </Slider>
</styles__GalleryContainer>
`;
